projecteru2/barrel
# Add ownership mechanism for a fixed-ip to record containers holding it

This PR is to introduce a mechanism to record containers for a fixed-ip which is borrowed by these container

What I did
1. Add a field called attr on fixed-ip meta, the field is descibled as below
```json
{
  "Attrs": {
    "Borrows": [
      {
        "ID": "ec07",
        "HostName": "dev-100.10.1.1",
      }
    ]
  }
}
```
2. When created a container which uses fixed-ip, record it on the attr of fixed-ip
3. When removing a container, remove the record of it on the attr of fixed-ip
4. Only when a fixed-ip has no borrowing, we can release it
5. In additional, introduced embed etcd and refactored etcd store impl to make unit test with embed etcd


- Verification
1. Created a fixed-ip container via barrel, and then start it
2. Stop the fixed-ip container
3. Create another fixed-ip container, and declare the same ip the previous container uses.
4. Start the second container
5. Remove the first container, the fixed-ip should not be released
6. Stop and remove the second container, the fixed-ip should be released
